### ISSUE TITLE
Exporting Configuration Profiles Intune Workload

### ISSUE THREAD

**Issue by alvs19** [microsoft/Microsoft365DSC#2198]
Hi Team,
Just wondering if configuration profiles should be exported as part of the INTUNE workload?

I have managed to get everything else exported / imported except for those.

Just wondering if it is a limitation of the tool or not.

**Comment by etec-masterofsynapse** [comment]
There are components present that would suggest that it can export these:
@(
"IntuneDeviceConfigurationPolicyAndroidDeviceOwner",
"IntuneDeviceConfigurationPolicyAndroidWorkProfile",
"IntuneDeviceConfigurationPolicyiOS",
"IntuneDeviceConfigurationPolicyWindows10"
)

Does it give you no output for these, do you receive an error, what happens?

Does it have the necessary permissions?

**Comment by alvs19** [comment]
As you can see the workload is only exporting the one policy
![image](https://user-images.githubusercontent.com/111731402/186290267-54b95b01-91c8-4ff3-a5a0-dad195b19f57.png)
There are a lot more than this in Endpoint manager.
![image](https://user-images.githubusercontent.com/111731402/186290329-b13df80d-34a6-45cf-9a58-6c5a08f05a54.png)

**Comment by etec-masterofsynapse** [comment]
I checked my export and I have no Configuration Profiles whatsoever in the resulting M365TenantConfig.ps1, even though I have 10 Profiles across all platforms in Endpoint Manager. There is definitely something wrong with the component script.

**Comment by alvs19** [comment]
Thanks for confirming and glad it isn't an issue I am just facing. Hopefully this can be resolved as the core configuration of our Intune consists in these profiles. 

Any ideas how we can get this resolved by the Devs?

**Comment by etec-masterofsynapse** [comment]
Same, my global Intune config is in these 10 policies as well.

For starters, we are in an issue thread within the official GitHub repo for the project, so that's a good place to be.
And from the other issues I get the feeling they are being worked on regularly. So lets have a little patience.

**Comment by Manbearpiet** [comment]
We have the same issues :(

**Comment by RuudGijsbers** [comment]
The custom policies and settings catalog policies are not configuration policies from a PowerShell / DSC perspective. I'm working on adding more resources to export. 

@Manbearpiet, are you using the latest version of Microsoft 365 DSC

**Comment by NikCharlebois** [comment]
As Ruud points out, we are not able to replicate empty exports being generated with the latest versions of the solution. Please update to the latest and let us know if the problem persists. Thanks

**Comment by andikrueger** [comment]
Closing due to inactivity.

**Comment by matt-resolve** [comment]
> The custom policies and settings catalog policies are not configuration policies from a PowerShell / DSC perspective. I'm working on adding more resources to export.
> 
> @Manbearpiet, are you using the latest version of Microsoft 365 DSC

@RuudGijsbers Hi, just wondering if there is an update on this?

**Comment by alvs19** [comment]
> > The custom policies and settings catalog policies are not configuration policies from a PowerShell / DSC perspective. I'm working on adding more resources to export.
> > @Manbearpiet, are you using the latest version of Microsoft 365 DSC
> 
> @RuudGijsbers Hi, just wondering if there is an update on this?

Hi,

I just tested with the latest version available on the PowerShell Gallery and can confirm that Configuration Profiles are still NOT being exported.

**Comment by andikrueger** [comment]
Did you remove all outdated dependencies?

you could run

update-M365DSCModule

or 

uninstall-m365dscoutdateddepenencies

**Comment by alvs19** [comment]
Hi Andi.

I did update the dependencies but I will remove the outdated ones and test
again and get back to you.

Cheers

**Comment by matt-resolve** [comment]
Hi Andi,

I have tried both updating the module along with uninstalling outdated dependencies just now, and unfortunately, the same issue whereby not all configuration profiles are being exported.

**Comment by andikrueger** [comment]
Which resource has the issues and could you share the export command as well as the generated configuration without any personal information? 

There are a few PRs on their way... Both contain Breaking changes and won't be release prior April 2023.

#2342
#2431

@William-Francillette Do you have an idea?

**Comment by William-Francillette** [comment]
At present and looking at the Intune portal, we have developed device restrictions and Wifi policies from the configuration profiles for all platforms. 
we also developed some of the resources from the Endpoint Security features.

We don't at present support the general custom setting catalog policies. Due to the large number of settings available we will need a strategy for those as 1000s of settings are available.
We could split the resources based on the CSP tree or the different categories/template - this would need to be assessed. 

More policies to comes in the future 

As Andy mentioned 2 resources will be updated in April due to breaking changes. The others mentioned should be working, they all have been tested and could export and update. If there is a specific issue, I'll be happy to look at it
